### PR TITLE
Replace all usermessages with net.

### PR DESCRIPTION
**Pull request by Aspect12**
Hi.

I have replaced all usermessages with net due to a recent exploit that a server I'm developing for has been getting hammered with. Not to mention umsg is deprecated, so this does no harm.

I've tested the addon in singleplayer and everything seems to be working fine, though feel free to bugtest too if you wish.

### CODE REVIEW

File: lua/weapons/gmod_tool/stools/advdupe2.lua [hunk 1]
```diff
@@ -269,12 +269,12 @@ if(SERVER)then
 		else --Non valid entity or clicked the world
 			if ply.AdvDupe2.Entities then
 				--clear the dupe
-				umsg.Start("AdvDupe2_RemoveGhosts", ply)
-				umsg.End()
+				net.Start("AdvDupe2_RemoveGhosts")
```
Comment (thegrb93) [inline lua/weapons/gmod_tool/stools/advdupe2.lua:272]: This is SERVER code so we should be sending to the client

File: lua/autorun/server/advdupe2_sv_init.lua [hunk 2]
```diff
@@ -17,12 +17,26 @@ AddCSLuaFile "advdupe2/sh_codec.lua"
...
+
 function AdvDupe2.Notify(ply,msg,typ, showsvr, dur)
-	umsg.Start("AdvDupe2Notify",ply)
-		umsg.String(msg)
-		umsg.Char(typ or NOTIFY_GENERIC)
-		umsg.Char(dur or 5)
-	umsg.End()
+	net.Start("AdvDupe2Notify")
+		net.WriteString(msg)
+		net.WriteFloat(typ or 0)
```
Comment (thegrb93) [inline lua/autorun/server/advdupe2_sv_init.lua:36]: WriteUInt(8, typ or 0) instead of WriteFloat

File: lua/autorun/client/advdupe2_cl_init.lua [hunk 3]
```diff
@@ -24,6 +24,6 @@ function AdvDupe2.Notify(msg,typ,dur)
 	//end
 end
 
-usermessage.Hook("AdvDupe2Notify",function(um)
-	AdvDupe2.Notify(um:ReadString(),um:ReadChar(),um:ReadChar())
+net.Receive("AdvDupe2Notify", function()
+	AdvDupe2.Notify(net.ReadString(), net.ReadFloat(), net.ReadFloat())
```
Comment (thegrb93) [inline lua/autorun/client/advdupe2_cl_init.lua:28]: ReadUInt(8) instead of ReadFloat

**Review by thegrb93 — CHANGES_REQUESTED** [review]

File: lua/autorun/server/advdupe2_sv_init.lua (hunk 2 above)
Comment (Divran) [reply lua/autorun/server/advdupe2_sv_init.lua:36]: it's value first, number of bits second https://wiki.facepunch.com/gmod/net.WriteUInt

**Review by Aspect12 — COMMENTED** [review]
Done - apologies for the SendToServer thing, not sure how that slipped up.

File: lua/weapons/gmod_tool/stools/advdupe2.lua [hunk 4]
```diff
@@ -380,15 +380,15 @@ if(SERVER)then
...
-				umsg.Short(areasize)
+			net.Start("AdvDupe2_CanAutoSave")
+				net.WriteVector(trace.HitPos)
+				net.WriteFloat(areasize)
 				if(trace.Entity)then
-					umsg.Short(trace.Entity:EntIndex())
+					net.WriteFloat(trace.Entity:EntIndex())
 				else
-					umsg.Short(0)
+					net.WriteFloat(0)
```
Comment (thegrb93) [inline lua/weapons/gmod_tool/stools/advdupe2.lua:389]: These two should be net.WriteUInt(x, 16), and the corresponding ReadFloats should be updated too.

File: lua/weapons/gmod_tool/stools/advdupe2.lua [hunk 5]
```diff
@@ -598,14 +598,8 @@ if(SERVER)then
 		if(not keep)then
 			ply.AdvDupe2.Name = nil
 		end
-		umsg.Start("AdvDupe2_ResetOffsets", ply)
-		umsg.End()
-	end
-
-	function AdvDupe2.UpdateProgressBar(ply, perc)
```
Comment (thegrb93) [inline lua/weapons/gmod_tool/stools/advdupe2.lua:605]: Isn't this function needed?

**Review by thegrb93 — CHANGES_REQUESTED** [review]

**Comment by thegrb93** [comment]
@Aspect12

File: lua/weapons/gmod_tool/stools/advdupe2.lua (hunk 5 above)
Comment (Aspect12) [reply lua/weapons/gmod_tool/stools/advdupe2.lua:605]: It was a duplicate. It exists right above.

**Review by thegrb93 — APPROVED** [review]

**Pull request merged**